src/main/java/uk/gov/hmcts/reform/sscs/service/hmc/topic/HearingUpdateService.java

```diff
@@ -97,4 +102,30 @@ public void setHearingStatus(String hearingId, @Valid SscsCaseData sscsCaseData,
 
         hearing.getValue().setHearingStatus(hearingStatus);
     }
+
+    public void setWorkBasketFields(String hearingId, @Valid SscsCaseData sscsCaseData, HmcStatus hmcStatus) {
+
+        WorkBasketFields workBasketFields = sscsCaseData.getWorkBasketFields();
+
+        if (isCaseListed(hmcStatus)) {
+            LocalDate hearingDate = getHearingDate(hearingId, sscsCaseData);
+            workBasketFields.setHearingDate(hearingDate);
+        } else {
+            workBasketFields.setHearingDate(null);
+        }
+
+    }
+
+    public LocalDate getHearingDate(String hearingId, @Valid SscsCaseData sscsCaseData) {
+        Hearing hearing = HearingsServiceHelper.getHearingById(Long.valueOf(hearingId), sscsCaseData);
+        if (nonNull(hearing) && nonNull(hearing.getValue().getStart())) {
+            return hearing.getValue().getStart().toLocalDate();
+        }
+        return null;
+    }
+
+    public boolean isCaseListed(HmcStatus hmcStatus) {
+        return LISTED.equals(hmcStatus)
+            || UPDATE_SUBMITTED.equals(hmcStatus);
```

Comment:
It's possible for a hearing to have an update sent on it whilst still waiting to be listed

Reply:
The Epims Id will be empty in this case so it will be set to null

Reply:
Can't you just remove the check then?